File: A2 - PastPaper Questions/Yearly/2023/9618_43_ON_23/VB/VowelCounter.vb

Function IterativeVowels(Value As String) As Integer
    Dim Total As Integer
    Dim FirstCharacter As String
    Total = 0
    While Len(Value) > 0
        FirstCharacter = Left(Value, 1)
        If InStr("aeiou", FirstCharacter) > 0 Then
            Total = Total + 1
        End If
        Value = Mid(Value, 2)
    Wend
    IterativeVowels = Total
End Function

Function RecursiveVowels(Value As String) As Integer
    If Len(Value) = 0 Then
        RecursiveVowels = 0
    Else
        If InStr("aeiou", Left(Value, 1)) > 0 Then
            RecursiveVowels = 1 + RecursiveVowels(Mid(Value, 2))
        Else
            RecursiveVowels = RecursiveVowels(Mid(Value, 2))
        End If
    End If
End Function

' Example calls
Sub Main()
    console.writeline(IterativeVowels("house"))
    console.writeline(Debug.Print RecursiveVowels("imagine"))
End Sub

File: A2 - PastPaper Questions/Yearly/2023/9618_43_ON_23/java/VowelCounter.java

public class VowelCounter {
    // Iterative method to count vowels
    public static int IterativeVowels(String value) {
        int total = 0;
        int lengthString = value.length();
        for (int x = 0; x < lengthString; x++) {
            char firstCharacter = value.charAt(0);
            if ("aeiou".indexOf(firstCharacter) >= 0) {
                total += 1;
            }
            value = value.substring(1);
        }
        return total;
    }

    // Recursive method to count vowels
    public static int RecursiveVowels(String value) {
        if (value.isEmpty()) {
            return 0;
        } else {
            char firstCharacter = value.charAt(0);
            if ("aeiou".indexOf(firstCharacter) >= 0) {
                return 1 + RecursiveVowels(value.substring(1));
            } else {
                return RecursiveVowels(value.substring(1));
            }
        }
    }

    public static void main(String[] args) {
        System.out.println(IterativeVowels("house")); // Example call for IterativeVowels
        System.out.println(RecursiveVowels("imagine")); // Example call for RecursiveVowels
    }
}
